open_seq2seq/encoders/resnet_encoder.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2018 NVIDIA Corporation
+from __future__ import absolute_import, division, print_function
+from __future__ import unicode_literals
+from six.moves import range
+
+import tensorflow as tf
+from .resnet_blocks import conv2d_fixed_padding, batch_norm, block_layer, \
+                           bottleneck_block_v1, bottleneck_block_v2, \
+                           building_block_v1, building_block_v2
+from .encoder import Encoder
+
+
+class ResNetEncoder(Encoder):
+  @staticmethod
+  def get_required_params():
+    return dict(Encoder.get_required_params(), **{
+
+    })
+
+  @staticmethod
+  def get_optional_params():
+    return dict(Encoder.get_optional_params(), **{
+
+    })
+
+  def __init__(self, params, model, name="resnet_encoder", mode='train'):
+    super(ResNetEncoder, self).__init__(params, model, name, mode)
+
+  def _encode(self, input_dict):
+    inputs = input_dict['source_tensors'][0]
+
+    self.resnet_size = 50
```

Comment:
Why these are hard-coded? Shouldn't we have them, and others, as optional parameters?